### ISSUE TITLE
Feature Request: better error messages when communicating

### ISSUE THREAD

**Issue by phmarek** [markusfisch/BinaryEye#149]
If there's an URL configured but then some problem with the server access happens, just some short note is displayed.

Please provide a more detailed error - be it "can't connect to server", "404", "403", or whatever.

Where could I see more details? A log viewer in Settings or so might be nice to have as well... 

I'm pondering whether the server's answer should be displayed -- but then it would have to be a (very) short "OK", "got it" or something like that.

**Comment by markusfisch** [comment]
You're right - the exact error condition or HTTP status code isn't displayed anywhere, unfortunately 😬  But of course I see how this can be useful!

So, there are some things we could do:
* always display the HTTP status code (with the returned content) in the Toast
* do that only for HTTP status codes < 200 and > 299
* show a snackbar instead of the Toast and have a "More" button that shows the details
* add a setting to toggle if the detailed error message should be shown or just the text as it is in case of failure

What would you prefer?

**Comment by phmarek** [comment]
Well, for a 200 it would be interesting to have the short answer visible - that could be the book title, in my use case.

For everything else a short sound signal (not vibration, that means "OK") and the http status, with a button to display the full result for diagnosis?

**Comment by markusfisch** [comment]
Okay, I'll try something like that.

**Comment by phmarek** [comment]
Thanks!

**Comment by markusfisch** [comment]
Implemented in https://github.com/markusfisch/BinaryEye/commit/37ea4ac9af3cf0935644687f7cbecf8edb62558b and https://github.com/markusfisch/BinaryEye/commit/35b6ead6626c807b7779c579152079540c419f36

Please check out version 1.37.0 which should become available on Google Play within the next few hours. Or you can try it directly from GitHub here: https://github.com/markusfisch/BinaryEye/releases/tag/1.37.0

**Comment by phmarek** [comment]
> Implemented in

Great, thanks!


> Please check out version 1.37.0

I'm using fdroid, but it'll be there soon too, I guess.

**Comment by markusfisch** [comment]
Yes, should be there tomorrow 👍

**Comment by phmarek** [comment]
Looks better!

A few points:

- When the phone is set to vibrate-only, I have error beeps. In silent mode (eg. in a library), it still produces a short squeak before shutting up. But I'm not sure whether you want to make these notifications configurable as well.... might be overkill. \\
	How about just some unique vibration patterns for "OK" and "not OK" - thrice short vs. one long or so?
- The error message is a bit truncated - `Failed to connect to /<ip>:<port>`, either remove the `/`, or add the `http` scheme in use, please.
- If the remote answers TCP but then gives no result (`netcat -l -p <port>`), there's no indication whether that worked or not... perhaps a 5 sec timeout or so would be in order?
- Is there another place than `logread` to find out what exactly went wrong? Not yet, right? Well, that might not be so important.

**Comment by markusfisch** [comment]
> When the phone is set to vibrate-only, I have error beeps. In silent mode (eg. in a library), it still produces a short squeak before shutting up. But I'm not sure whether you want to make these notifications configurable as well.... might be overkill. \
How about just some unique vibration patterns for "OK" and "not OK" - thrice short vs. one long or so?

Well, there should already be a different haptic feedback for the error and the success vibration (see https://github.com/markusfisch/BinaryEye/blob/master/app/src/main/kotlin/de/markusfisch/android/binaryeye/activity/CameraActivity.kt#L712). At least on my devices I do feel a difference 🤔 

> The error message is a bit truncated - Failed to connect to /<ip>:<port>, either remove the /, or add the http scheme in use, please.

The error message _is_ the message of the exception, actually 😉 (see https://github.com/markusfisch/BinaryEye/blob/master/app/src/main/kotlin/de/markusfisch/android/binaryeye/net/ScanSender.kt#L100)

> If the remote answers TCP but then gives no result (netcat -l -p <port>), there's no indication whether that worked or not... perhaps a 5 sec timeout or so would be in order?

Yes, good idea! 👍 Added in https://github.com/markusfisch/BinaryEye/commit/d594a43d6d113f1a57748aa4761b4e8e1777faf3

Will be in the next version.

> Is there another place than logread to find out what exactly went wrong? Not yet, right? Well, that might not be so important.

No, currently not. But since the app is already showing the exception message, there's probably not much more to see 🤔
